### PR TITLE
feat: add account deletion flow

### PR DESCRIPTION
**Pull request by nickytonline**
## Description

Adds the account deletion flow to the app. The only way you can test this is by deleting your own account in beta. Once tested, your soft delete can be reset, but it's a manual process.

To test this, head to https://deploy-preview-1990--oss-insights.netlify.app/user/settings and scroll to the bottom to delete your account.

## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents

Closes https://github.com/open-sauced/engineering/issues/174
Relates to https://github.com/open-sauced/api/pull/396
fixes #1990

## Mobile & Desktop Screenshots/Recordings

**Desktop**

CleanShot 2023-10-25 at 09 50 44

**Smaller Screens**

CleanShot 2023-10-25 at 07 31 22

## Added tests?

- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [ ] 🙅 no documentation needed

### CODE REVIEW

**Comment by bdougie** [comment]
@nickytonline are we able to add a "Type CONFIRM" to confirm, similar to deleting insight pages?

Also can you share what is blocking this here?

**Comment by nickytonline** [comment]
> @nickytonline are we able to add a "Type CONFIRM" to confirm, similar to deleting insight pages?
> 
> Also can you share what is blocking this here?

We can type CONFIRM. I'll go grab that from the insights code.

The blocker is I need to build the API endpoint which I'm looking at shortly. That has to go in first before this is merged.

File: middleware.ts [hunk 1]
```diff
@@ -28,6 +29,14 @@ export async function middleware(req: NextRequest) {
     data: { session },
   } = await supabase.auth.getSession();
 
+  if (session?.user && req.nextUrl.pathname === "/account-deleted") {
+    // Delete the account from Supabase and log the user out.
+    await supabase.auth.admin.deleteUser(session.user.id);
```
Comment (nickytonline) [inline middleware.ts:34]: Just wanted to check if this is cool to run @brandonroberts @jpmcb or is there a soft delete we want here as well?

[the remark this answers is not in the document]
Comment (brandonroberts) [reply middleware.ts:34]: I think this is the way to go as far as Supabase goes

File: pages/account-deleted.tsx [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import TopNav from "components/organisms/TopNav/top-nav";
+
+const AccountDeletedPage = () => {
+  return (
+    <div className="min-h-screen flex flex-col">
+      <TopNav />
+      <main className="page-container flex flex-col m-4 pt-28 items-center">
+        <div className="info-container container w-full">
+          <h1 className="text-2xl">Account Deleted</h1>
+          <p>Your account has been deleted.</p>
```
Comment (brandonroberts) [inline pages/account-deleted.tsx:10]: Should we do a hard reload and redirect them back to `/` after coming here?

File: middleware.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (jpmcb) [reply middleware.ts:34]: Makes sense to me: If they come back I believe their ID and user info from Github will propegate back through and still align 👍🏼

**Review by jpmcb — APPROVED** [review]
LGTM! Nice work Nick!!

**Review by OgDev-01 — APPROVED** [review]
🍕

File: pages/account-deleted.tsx (hunk 2 above)
Comment (nickytonline) [reply pages/account-deleted.tsx:10]: Sounds good. On it!

Comment (nickytonline) [reply pages/account-deleted.tsx]: Done in 2d7c6bcb9f9db6b263c66c9b114a64cc4b63a4fb

**Comment by bdougie** [comment]
Want to comment earlier before this was merged but the UI for the danger zone, same as insights with the a grey background 

image

**Pull request merged**